File: lib/breadth-first-search.py
from collections import deque

graph = {}
graph['you'] = ['Easton', 'Arthur', 'Injoong']
graph['Easton'] = ['Kevin', 'Sarah']
graph['Arthur'] = ['Marty', 'Robyn']
graph['Injoong'] = ['Mariah', 'Antreo']
graph['Kevin'] = []
graph['Sarah'] = []
graph['Marty'] = []
graph['Robyn'] = []
graph['Mariah'] = []
graph['Antreo'] = []

def search(name):
    search_queue = deque()
    search_queue += graph[name]
    searched = []

    def person_is_seller(name):
        return name[-1] == 'h'

    while search_queue:
        person = search_queue.popleft()
        if not person in searched:
            if(person_is_seller(person)):
                print(person + 'is a mango seller!')
                return True
            else:
                search_queue += graph[person]
                searched.append(person)
    return False

File: lib/breadth-first-search.js
const graph = {}
graph['you'] = ['Easton', 'Arthur', 'Injoong']
graph['Easton'] = ['Kevin', 'Sarah']
graph['Arthur'] = ['Marty', 'Robyn']
graph['Injoong'] = ['Mariah', 'Antreo']
graph['Kevin'] = []
graph['Sarah'] = []
graph['Marty'] = []
graph['Robyn'] = []
graph['Mariah'] = []
graph['Antreo'] = []

module.exports = breadthFirstSearch = name => {
    //using the last letter of their name is just a simple way to test the algorithm
    const personIsSeller = _name => _name[_name.length - 1] === 'h';
    let search_queue = [], searched = {};
    search_queue.push(...graph[name]);
    
    while(search_queue.length) {
        let person = search_queue.splice(0, 1)[0];
        if(!searched[person]) {
            if(personIsSeller(person)) {
                return {
                    found: true,
                    name: person
                };
            }
            else {
                search_queue.push(...graph[person]);
                searched[person] = true;
            }
        }
    }
    return {
        found: false,
        name: null
    };
}